GlobusApp login, logout, and login_required (#1041)
globus/globus-sdk-python: docs/experimental/examples/oauth2/globus_app.rst

Manually Running Login Flows
----------------------------

> While your app will automatically initiate and oversee login flows as detected,

Reviewer: ```suggestion
While your app will automatically initiate and oversee login flows when needed,
```

sometimes an author may want to explicitly control when an authorization occurs. To
manually trigger a login flow, call ``GlobusApp.login(...)``. The app will evaluate the
current scope requirements against available tokens, initiating a login flow if it
determines that any requirements across any resource servers are unmet. Resulting tokens
will be cached for future use.

This method accepts two optional keyword args:

> -   ``auth_params`` a collection of additional auth parameters to customize the login.

Reviewer: ```suggestion
-   ``auth_params``, a collection of additional auth parameters to customize the login.
```

>     This allows for specifications such as, requiring that a user be logged in with an

Reviewer: ```suggestion
    This allows for specifications such as requiring that a user be logged in with an
```

>     MFA token or rendering the authorization webpage with a specific message.

Author: I'd love to link out to another doc here expanding on what can be passed but the two candidates I looked at were:
* [SDK Docs on GAREs](https://globus-sdk-python.readthedocs.io/en/stable/experimental/auth_requirements_errors.html) which don't enumerate the list of valid options.
* [Auth API Docs on Authorize Code Grant](https://docs.globus.org/api/auth/reference/#authorization_code_grant_preferred) which includes additional items beyond those that go strictly into a GAREData object.

> -   ``force``, a boolean flag instruct the app to run a login flow even if it believes
>     that no such login flow is required.

Reviewer: ```suggestion
-   ``force``, a boolean flag instructing the app to perform a login flow
    regardless of whether it is required.
```
